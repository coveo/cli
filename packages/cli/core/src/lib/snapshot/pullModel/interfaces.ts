import {ResourceSnapshotExportConfigurationModel} from '@coveo/platform-client';

export type SnapshotPullModelResources =
  ResourceSnapshotExportConfigurationModel['resourcesToExport'];

export interface SnapshotPullModel {
  resourcesToExport: SnapshotPullModelResources;
  orgId?: string;
  includeChildrenResources?: boolean;
}
