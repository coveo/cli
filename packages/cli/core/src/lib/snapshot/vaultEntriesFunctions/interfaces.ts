import type {Configuration} from '@coveo/cli-commons/lib/config/config';
import type {Snapshot} from '../snapshot';
import type {SnapshotReporter} from '../snapshotReporter';

export type VaultTransferFunctionsParam = {
  reporter: SnapshotReporter;
  snapshot: Snapshot;
  cfg: Configuration;
  projectPath?: string;
};
