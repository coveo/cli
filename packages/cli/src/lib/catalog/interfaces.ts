import {MetadataValue} from '@coveo/push-api-client';
import {CreateCatalogConfigurationModel} from '@coveord/platform-client';

export type MetadataKey = string;
export type MapValue = string | number | null;

/**
 * A map which stores the medata key as well as all the possible values for that metada
 */
export type MetadataValueMap = Map<MetadataKey, MapValue[]>;

export interface CatalogFieldStrutureValue {
  possibleIdFields: MetadataKey[];
  objectType: MetadataValue;
}

export interface CatalogFieldStruture {
  product: CatalogFieldStrutureValue;
  variant?: CatalogFieldStrutureValue;
  availability?: CatalogFieldStrutureValue & {
    availableSkusField: MetadataValue;
  };
}

export type PartialCatalogConfigurationModel = Pick<
  CreateCatalogConfigurationModel,
  'product' | 'variant' | 'availability'
>;
