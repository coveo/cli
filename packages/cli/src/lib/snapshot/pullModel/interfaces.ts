import {ResourceSnapshotType} from '@coveord/platform-client';

export type SnapshotPullModelResourceType =
  | ResourceSnapshotType.extension
  | ResourceSnapshotType.field
  | ResourceSnapshotType.mlModel
  | ResourceSnapshotType.queryPipeline
  | ResourceSnapshotType.searchPage
  | ResourceSnapshotType.source
  | ResourceSnapshotType.subscription;

export const SnapshotPullModelResourceTypes: SnapshotPullModelResourceType[] = [
  ResourceSnapshotType.extension,
  ResourceSnapshotType.field,
  ResourceSnapshotType.mlModel,
  ResourceSnapshotType.queryPipeline,
  ResourceSnapshotType.searchPage,
  ResourceSnapshotType.source,
  ResourceSnapshotType.subscription,
];

type SnapshotPullModelResources = Partial<
  Record<SnapshotPullModelResourceType, Array<string>>
>;

export interface SnapshotPullModel {
  resources: SnapshotPullModelResources;
  options?: {
    includeChildrenResources?: boolean;
  };
}
