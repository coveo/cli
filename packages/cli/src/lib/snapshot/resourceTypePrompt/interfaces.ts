export enum ResourceTypeActions {
  Add = 'add',
  Skip = 'skip',
  Delete = 'delete',
  Edit = 'edit',
}

export interface ResourceTypePromptState {
  longestKeyDisplayNameLength: number;
}
