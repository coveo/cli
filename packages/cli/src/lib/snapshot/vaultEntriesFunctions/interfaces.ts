import type {Configuration} from '../../config/config';
import type {Snapshot} from '../snapshot';
import type {SnapshotReporter} from '../snapshotReporter';

export type VaultTransferFunctionsParam = {
  reporter: SnapshotReporter;
  snapshot: Snapshot;
  cfg: Configuration;
  projectPath?: string;
};
