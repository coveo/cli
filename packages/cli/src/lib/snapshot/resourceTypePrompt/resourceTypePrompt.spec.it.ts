import {getSelectResourceTypesPrompt, ResourcePromptDefaults} from './index';
import {fancy} from 'fancy-test';
import {MockSTDIN, stdin} from 'mock-stdin';
import {ResourceTypeActions} from './interfaces';
import {SnapshotPullModelResourceType} from '../pullModel/interfaces';
import {ResourceSnapshotType} from '@coveord/platform-client';

describe('ResourceTypePrompt - Integration Test', () => {
  const defaultAnswers: [SnapshotPullModelResourceType, ResourceTypeActions][] =
    [
      [ResourceSnapshotType.extension, ResourceTypeActions.Skip],
      [ResourceSnapshotType.field, ResourceTypeActions.Skip],
      [ResourceSnapshotType.mlModel, ResourceTypeActions.Skip],
      [ResourceSnapshotType.queryPipeline, ResourceTypeActions.Skip],
      [ResourceSnapshotType.searchPage, ResourceTypeActions.Skip],
      [ResourceSnapshotType.source, ResourceTypeActions.Skip],
      [ResourceSnapshotType.subscription, ResourceTypeActions.Skip],
    ];

  const arrowKeys = {
    up: '\x1b[A',
    down: '\x1b[B',
    left: '\x1b[C',
    right: '\x1b[D',
  };

  /**
   * ANSI Escape Sequence - Cursor Horizontal Absolute
   * https://vt100.net/docs/vt510-rm/CHA.html
   * ECMA-48 5th Edition, Section 8.3.9
   * In non-beepboop language: It move the cursor at the beginning of the line.
   */
  const ansiCursorHorizontalAbsolute = '\x1b[G';

  const fancyIt = () =>
    fancy.env({FORCE_COLOR: '3'}).stdout({stripColor: false}).it;
  let mockedStdin: MockSTDIN;

  beforeEach(() => {
    mockedStdin = stdin();
  });

  afterEach(() => {
    mockedStdin.restore();
  });

  describe('when started', () => {
    fancyIt()(
      'should display the question and the first 7 resources',
      (output) => {
        getSelectResourceTypesPrompt('someQuestion');

        expect(output.stdout).toMatchSnapshot();

        mockedStdin.end();
      }
    );
  });

  describe.each([
    ['a', ResourceTypeActions.Add, {}, {EXTENSION: ResourceTypeActions.Skip}],
    ['s', ResourceTypeActions.Skip, {}, null],
    [
      'd',
      ResourceTypeActions.Delete,
      {EXTENSION: ResourceTypeActions.Skip},
      {},
    ],
    ['e', ResourceTypeActions.Edit, {EXTENSION: ResourceTypeActions.Skip}, {}],
  ])(
    'when pressing %s',
    (
      key: string,
      associatedAction: ResourceTypeActions,
      compatibleDefault: ResourcePromptDefaults,
      incompatibleDefault: ResourcePromptDefaults | null
    ) => {
      let answersWithAssociatedAction: [
        SnapshotPullModelResourceType,
        ResourceTypeActions
      ][];

      beforeAll(() => {
        answersWithAssociatedAction = [...defaultAnswers];
        answersWithAssociatedAction.shift();
        answersWithAssociatedAction.unshift([
          ResourceSnapshotType.extension,
          associatedAction,
        ]);
      });

      fancyIt()(
        `should select ${associatedAction} for the highlighted resource if possible`,
        async (output) => {
          const promptPromise = getSelectResourceTypesPrompt(
            'someQuestion',
            compatibleDefault
          );
          const preStdoutLength = output.stdout.length;
          mockedStdin.send(key);
          await Promise.resolve();

          expect(output.stdout.substr(preStdoutLength)).toMatchSnapshot();
          mockedStdin.send('\n');
          await expect(promptPromise).resolves.toEqual({
            someQuestion: answersWithAssociatedAction,
          });
          mockedStdin.end();
        }
      );

      fancyIt()(
        `should do nothing if the resource is already associated to ${associatedAction}`,
        async (output) => {
          const promptPromise = getSelectResourceTypesPrompt('someQuestion', {
            EXTENSION: associatedAction,
          });
          const preInputStdout = output.stdout;

          mockedStdin.send(key);
          await Promise.resolve();

          const postInputStdout = output.stdout.split(
            ansiCursorHorizontalAbsolute
          )[1];

          mockedStdin.send('\n');

          expect(postInputStdout).toEqual(preInputStdout);
          await expect(promptPromise).resolves.toEqual({
            someQuestion: answersWithAssociatedAction,
          });
          mockedStdin.end();
        }
      );

      if (incompatibleDefault) {
        fancyIt()(
          'should do nothing if the resource/operation association is invalid',
          async (output) => {
            const promptPromise = getSelectResourceTypesPrompt(
              'someQuestion',
              incompatibleDefault
            );
            const preInputStdout = output.stdout;

            mockedStdin.send(key);
            await Promise.resolve();

            const postInputStdout = output.stdout.split(
              ansiCursorHorizontalAbsolute
            )[1];

            mockedStdin.send('\n');

            expect(postInputStdout).toEqual(preInputStdout);
            await expect(promptPromise).resolves.toEqual({
              someQuestion: defaultAnswers,
            });
            mockedStdin.end();
          }
        );
      }
    }
  );

  //#region Refactor maybe
  describe.each([
    ['up', arrowKeys.up, 6],
    ['down', arrowKeys.down, 6],
    ['left', arrowKeys.left, 3],
    ['right', arrowKeys.right, 3],
  ])(
    'when pressing %s arrow key',
    (_direction, asciiCode, nbKeyPressBeforeLooping) => {
      fancyIt()(
        'should select the next valid value/key if there is one',
        async (output) => {
          getSelectResourceTypesPrompt('someQuestion');

          for (let i = 0; i < nbKeyPressBeforeLooping; i++) {
            mockedStdin.send(asciiCode);
          }
          await Promise.resolve();

          const preInputStdout = output.stdout
            .split(ansiCursorHorizontalAbsolute)
            .pop();

          mockedStdin.send(asciiCode);
          await Promise.resolve();

          const postInputStdout = output.stdout
            .split(ansiCursorHorizontalAbsolute)
            .pop();

          expect(postInputStdout).not.toEqual(preInputStdout);
          expect(postInputStdout).toMatchSnapshot();

          mockedStdin.end();
        }
      );

      fancyIt()(
        'should loop and select the next valid value/key',
        async (output) => {
          getSelectResourceTypesPrompt('someQuestion');

          for (let i = 0; i < nbKeyPressBeforeLooping; i++) {
            mockedStdin.send(asciiCode);
          }
          await Promise.resolve();

          const preInputStdout = output.stdout
            .split(ansiCursorHorizontalAbsolute)
            .pop();

          mockedStdin.send(asciiCode);
          await Promise.resolve();

          const postInputStdout = output.stdout
            .split(ansiCursorHorizontalAbsolute)
            .pop();

          expect(postInputStdout).not.toEqual(preInputStdout);
          expect(postInputStdout).toMatchSnapshot();

          mockedStdin.end();
        }
      );
    }
  );

  describe('when pressing space bar', () => {
    fancyIt()(
      'should do nothing if the highlighted key/value combination is disabled',
      async (output) => {
        const promptPromise = getSelectResourceTypesPrompt('someQuestion');
        const preInputStdout = output.stdout;

        mockedStdin.send(' ');
        await Promise.resolve();

        const postInputStdout = output.stdout.split(
          ansiCursorHorizontalAbsolute
        )[1];

        expect(postInputStdout).toEqual(preInputStdout);
        expect(postInputStdout).toMatchSnapshot();

        mockedStdin.end();
        await promptPromise;
      }
    );

    fancyIt()(
      'should select if the highlighted key/value combination is not disabled',
      async (output) => {
        const promptPromise = getSelectResourceTypesPrompt('someQuestion', {});
        const preInputStdout = output.stdout;

        mockedStdin.send(' ');
        await Promise.resolve();

        const postInputStdout = output.stdout.split(
          ansiCursorHorizontalAbsolute
        )[1];

        expect(postInputStdout).not.toEqual(preInputStdout);
        expect(postInputStdout).toMatchSnapshot();

        mockedStdin.end();
        await promptPromise;
      }
    );
  });

  describe('when pressing enter', () => {
    fancyIt()(
      'should resolves the promise with the returned values',
      async () => {
        const promptPromise = getSelectResourceTypesPrompt('someQuestion');

        mockedStdin.send('\n');
        await expect(promptPromise).resolves.toEqual({
          someQuestion: defaultAnswers,
        });
        mockedStdin.end();
      }
    );
  });
});
