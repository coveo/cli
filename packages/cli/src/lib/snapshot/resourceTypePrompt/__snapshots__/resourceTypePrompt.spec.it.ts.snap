// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceTypePrompt - Integration Test when pressing a should select add for the highlighted resource if possible 1`] = `
"[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [4m[7m[32m[1mA[22mdd[39m[27m[24m [97m[1mS[22mkip[39m [90mDelete[39m [90mEdit[39m
 FIELD         	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 ML_MODEL      	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 QUERY_PIPELINE	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SEARCH_PAGE   	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SOURCE        	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SUBSCRIPTION  	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing d should select delete for the highlighted resource if possible 1`] = `
"[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [97m[1mS[22mkip[39m [4m[31m[1mD[22melete[39m[24m [36m[1mE[22mdit[39m
 FIELD         	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 ML_MODEL      	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 QUERY_PIPELINE	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SEARCH_PAGE   	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SOURCE        	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SUBSCRIPTION  	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing down arrow key should loop and select the next valid value/key 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 [2m*[22m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing down arrow key should select the next valid value/key if there is one 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 [2m*[22m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing e should select edit for the highlighted resource if possible 1`] = `
"[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [97m[1mS[22mkip[39m [31m[1mD[22melete[39m [4m[36m[1mE[22mdit[39m[24m
 FIELD         	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 ML_MODEL      	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 QUERY_PIPELINE	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SEARCH_PAGE   	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SOURCE        	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SUBSCRIPTION  	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing left arrow key should loop and select the next valid value/key 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SEARCH_PAGE   	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing left arrow key should select the next valid value/key if there is one 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SEARCH_PAGE   	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing right arrow key should loop and select the next valid value/key 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SEARCH_PAGE   	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing right arrow key should select the next valid value/key if there is one 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SEARCH_PAGE   	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing s should select skip for the highlighted resource if possible 1`] = `
"[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[32m[1mA[22mdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 FIELD         	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 ML_MODEL      	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 QUERY_PIPELINE	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SEARCH_PAGE   	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SOURCE        	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SUBSCRIPTION  	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing space bar should do nothing if the highlighted key/value combination is disabled 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SEARCH_PAGE   	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing space bar should select if the highlighted key/value combination is not disabled 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [4m[7m[32m[1mA[22mdd[39m[27m[24m [97m[1mS[22mkip[39m [90mDelete[39m [90mEdit[39m
 FIELD         	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 ML_MODEL      	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 QUERY_PIPELINE	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SEARCH_PAGE   	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SOURCE        	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
 SUBSCRIPTION  	 [32m[1mA[22mdd[39m [4m[97m[1mS[22mkip[39m[24m [90mDelete[39m [90mEdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing up arrow key should loop and select the next valid value/key 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 [2m*[22m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when pressing up arrow key should select the next valid value/key if there is one 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 [2m*[22m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;

exports[`ResourceTypePrompt - Integration Test when started should display the question and the first 7 resources 1`] = `
"[32m?[39m [1msomeQuestion:[22m[0m [0m
[36m‚ùØ[39mEXTENSION     	 [7m[90mAdd[39m[27m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 FIELD         	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 ML_MODEL      	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 QUERY_PIPELINE	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SEARCH_PAGE   	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SOURCE        	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
 SUBSCRIPTION  	 [90mAdd[39m [4m[97m[1mS[22mkip[39m[24m [31m[1mD[22melete[39m [36m[1mE[22mdit[39m
[2m(Move up and down to reveal more choices)[22m[41D[41C"
`;
